### PR TITLE
CEM-2061 box composition 2

### PR DESCRIPTION
**Pull request by roggc**
bc2

### CODE REVIEW

File: src/Containers/BoxComposition2/BoxComposition2.tsx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    useEffect(()=>{
+        setTimeout(()=>{
+            setShowBox2(true)
+        },1000)
+        setTimeout(()=>{
+            setShowBox3(true)
+        },2000)
+    },[])
+    return     (
+        <div>
```
Comment (ralph-dev) [inline src/Containers/BoxComposition2/BoxComposition2.tsx:31]: where is the ...props spread here?

File: src/Containers/BoxComposition2/BoxComposition2.tsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
+import React,{useState,useEffect} from 'react'
+import styled from 'styled-components'
+
```
Comment (ralph-dev) [inline src/Containers/BoxComposition2/BoxComposition2.tsx:3]: You need to extend div for the root prop

File: src/Containers/BoxComposition2/BoxComposition2.tsx [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+            <Box1 {...box1} />
+            {showBox2&&<Box2 {...box2} />}
+            {showBox3&&<Box3 {...box3} />}
+        </div>
+    )
+}
+
+const Box3:React.FC<IBoxProps>=({imgSrc,top,left,width,height,borderRadius}):React.ReactElement=>
+    (
+        <AnimatedDiv3 top={top} left={left}>
```
Comment (ralph-dev) [inline src/Containers/BoxComposition2/BoxComposition2.tsx:41]: BAD VARIABLE NAMES 
AnimatedDiv3 is not good!!

File: src/Containers/BoxComposition2/BoxComposition2.tsx [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    )
+
+const Box1:React.FC<IBoxProps>=({imgSrc,top,left,width,height,borderRadius}):React.ReactElement=>
+    (
+        <AnimatedDiv1 top={top} left={left}>
+            <Img src={imgSrc} width={width} height={height} borderRadius={borderRadius} />
+        </AnimatedDiv1>
+    )
+
+interface IDivProps{
```
Comment (ralph-dev) [inline src/Containers/BoxComposition2/BoxComposition2.tsx:60]: Can't this be replaced with HTMLDivProps?

**Review by ralph-dev — CHANGES_REQUESTED** [review]
Code Quality is very bad, name your variables properly. On the bright side the Architecture is correct!

File: src/Containers/BoxComposition2/BoxComposition2.tsx (hunk 3 above)
Comment (roggc) [reply src/Containers/BoxComposition2/BoxComposition2.tsx:41]: ThirdAnimatedDiv will be ok?

File: src/Containers/BoxComposition2/BoxComposition2.tsx (hunk 4 above)
Comment (roggc) [reply src/Containers/BoxComposition2/BoxComposition2.tsx:60]: no, I don't think so...

File: src/Containers/BoxComposition2/BoxComposition2.tsx (hunk 1 above)
Comment (roggc) [reply src/Containers/BoxComposition2/BoxComposition2.tsx:31]: I don't understand...

File: src/Containers/BoxComposition2/BoxComposition2.tsx (hunk 3 above)
Comment (ralph-dev) [reply src/Containers/BoxComposition2/BoxComposition2.tsx:41]: No. The name has no MEANING.
What is this div?
Is this the CenterDiv
TopLeftDiv etc?
You should be able to read the code and instantly understand what relates to what

File: src/Containers/BoxComposition2/BoxComposition2.tsx (hunk 1 above)
Comment (ralph-dev) [reply src/Containers/BoxComposition2/BoxComposition2.tsx:26]: You need to do
<div {...props} />
That way when people use this component they can add a margin to it!

**Review by ralph-dev — APPROVED** [review]
Looks a lot better just spread the correct props of the root container so another person can add margin when using this componnent

**Comment by ralph-dev** [comment]
You forgot to extend HTMLDIVPROPS to the Prop now that you have ...props btw

**Pull request merged**